test/protocol/eval_test.rb

```diff
@@ -17,9 +17,9 @@ def test_eval_evaluates_arithmetic_expressions
       run_protocol_scenario PROGRAM do
         req_add_breakpoint 5
         req_continue
-        assert_repl_result 2, expression: 'a'
-        assert_repl_result 4, expression: 'd'
-        assert_repl_result 3, expression: '1+2'
+        assert_repl_result({value: '2', type: 'Integer'}, expression: 'a')
```

Comment:
Why don't we make `value` and `type` keyword arguments too? This 2 level hash will make it harder to write and seems unnecessary.

Reply:
Because assertion methods are usually `assert_something expected, result`.

Reply:
I don't think that's a strict rule that we must follow. And in this case we don't even pass the `result` in anyway, we just tell the helper how to get the result. 
So I feel that following that rule doesn't get us much benefit in terms of readability improvement but instead making things more complicated than it needs to be.

Reply:
I want to follow that rule because it's easy for users to configure the expected result as a first argument.
I think that ` assert_repl_result({value: '2', type: 'Integer'}, 'a')` is the best. We can remove 2 level hash in this case.

Reply:
ok let's give it a try 👍